### DISCUSSION TITLE
Toggle Run

### DISCUSSION THREAD

**Discussion by Trigger-1** [FakeFishGames/Barotrauma#5209] [Ideas and suggestions]
Let me preface by saying that I love this game. The chaos of playing with fandoms is hilarious, the co-op is fantastic, and the constant content updates keep me coming back.

However, as someone with arthritis, holding the Run button down constantly takes its toll. An option to toggle Run would be a fantastic quality of life upgrade, especially if tapping the button while moving activated Run, and automatically switched back to walking after releasing the movement keys.

**Comment by AzureNocte** [comment]
Indeed

**Comment by BlackMoons** [comment]
Agreed, it literally hurts having to hold shift for hours at a time while playing and almost nobody ever walks anywhere, and my hands are otherwise rather normal. 

Id go so far as to just have a 'Normally run' checkbox even, and people can hold shift to walk if its enabled.

**Comment by Tsunder** [comment]
valve please add this!

we run 99% of the time we move, holding shift for 99% is just bad!

just about every other game w/ run I can think of has auto-run and toggle run by default!

**Comment by lIllIIlI** [comment]
> Created in 2021

amazing
toggle bind to capslock or invert it to shift=walk

**Comment by Tsunder** [comment]
#14883  after three, almost four years

**Comment by Regalis11** [comment]
Implemented in https://github.com/FakeFishGames/Barotrauma-development/pull/5922 (PR in our private repo), coming up in a future update!
